dev/com.ibm.ws.concurrent_fat_reactive/test-applications/reactiveapp/src/reactiveapp/web/ConcurrentReactiveServlet.java

```diff
@@ -12,35 +12,111 @@
  *******************************************************************************/
 package reactiveapp.web;
 
-import javax.naming.InitialContext;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 
 import org.junit.Test;
 
 import componenttest.app.FATServlet;
 import jakarta.annotation.Resource;
+import jakarta.enterprise.concurrent.ContextService;
 import jakarta.enterprise.concurrent.ManagedExecutorService;
 import jakarta.servlet.annotation.WebServlet;
 
 @SuppressWarnings("serial")
 @WebServlet("/ConcurrentReactiveServlet")
 public class ConcurrentReactiveServlet extends FATServlet {
+    private static final long TIMEOUT_NS = TimeUnit.MINUTES.toNanos(5);
 
     @Resource(name = "java:comp/env/concurrent/executorRef")
     private ManagedExecutorService executor;
 
-    /*
+    @Resource
+    private ContextService contextSvcDefault;
+
+    /**
      * Test that context is available at different points in a flow (publisher, subscriber)
      * when using a ManagerExecutor.
      */
     @Test
     public void basicFlowTest() throws Exception {
+        final ContextCDLImpl continueLatch = new ContextCDLImpl(1);
+
+        ThreadPublisher publisher = new ThreadPublisher(executor);
+        ThreadSubscriber subscriber = new ThreadSubscriber();
+        publisher.subscribe(subscriber);
+        publisher.offer(continueLatch, null);
 
-        new InitialContext().lookup("java:comp/env/entry1");
+        if (publisher.getClosedException() != null)
+            fail("Context was not available in Publisher");
+        if (subscriber.getClosedException() != null)
+            fail("Context was not available in Subscriber");
```

Comment:
It will be helpful to the developer who is debugging issues to see the exception stack in the test failure.
```suggestion
        if (publisher.getClosedException() != null)
            throw (AssertionError)
                new AssertionError("Context was not available in Publisher")
                    .initCause(publisher.getClosedException());
        if (subscriber.getClosedException() != null)
            throw (AssertionError)
                new AssertionError("Context was not available in Subscriber")
                    .initCause(subscriber.getClosedException());
```